src/kakarot/accounts/library.cairo

```diff
@@ -160,198 +154,163 @@ namespace AccountContract {
-            with_attr error_message("EOAs cannot have code") {
-                assert bytecode_len = 0;
-            }
 
-            return ();
+        with_attr error_message("Incorrect signature length") {
+            assert signature_len = 5;
         }
+        let r = Uint256(signature[0], signature[1]);
+        let s = Uint256(signature[2], signature[3]);
+        let v = signature[4];
 
-        let (address) = Account_evm_address.read();
-        let (tx_info) = get_tx_info();
-
-        // Assert signature field is of length 5: r_low, r_high, s_low, s_high, v
-        assert tx_info.signature_len = 5;
-        let r = Uint256(tx_info.signature[0], tx_info.signature[1]);
-        let s = Uint256(tx_info.signature[2], tx_info.signature[3]);
-        let v = tx_info.signature[4];
-        let (_, chain_id) = unsigned_div_rem(tx_info.chain_id, 2 ** 32);
-
-        // Unpack the tx data
-        let packed_tx_data_len = [call_array].data_len;
-        let packed_tx_data = calldata + [call_array].data_offset;
-
-        let tx_data_len = [packed_tx_data];
-        let (tx_data) = Helpers.load_packed_bytes(
-            packed_tx_data_len - 1, packed_tx_data + 1, tx_data_len
-        );
-
-        Internals.validate(address, tx_info.nonce, chain_id, r, s, v, tx_data_len, tx_data);
-
-        validate(
-            call_array_len=call_array_len - 1,
-            call_array=call_array + CallArray.SIZE,
-            calldata_len=calldata_len,
-            calldata=calldata,
-        );
-
-        return ();
-    }
-
-    // @notice Execute the transaction.
-    // @param call_array_len The length of the call array.
-    // @param call_array The call array.
-    // @param calldata_len The length of the calldata.
-    // @param calldata The calldata.
-    // @param response The response data array to be updated.
-    // @return response_len The total length of the response data array.
-    func execute{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        bitwise_ptr: BitwiseBuiltin*,
-        range_check_ptr,
-    }(
-        call_array_len: felt,
-        call_array: CallArray*,
-        calldata_len: felt,
-        calldata: felt*,
-        response: felt*,
-    ) -> (response_len: felt) {
-        alloc_locals;
-        if (call_array_len == 0) {
-            return (response_len=0);
+        let tx_type = EthTransaction.get_tx_type(tx_data);
+        local y_parity: felt;
+        local pre_eip155_tx;
+        if (tx_type == 0) {
+            let is_eip155_tx = is_le(v, 28);
+            assert pre_eip155_tx = is_eip155_tx;
+            if (is_eip155_tx != FALSE) {
+                assert y_parity = v - 27;
+            } else {
+                assert y_parity = (v - 2 * chain_id - 35);
+            }
+            tempvar range_check_ptr = range_check_ptr;
+        } else {
+            assert pre_eip155_tx = FALSE;
+            assert y_parity = v;
+            tempvar range_check_ptr = range_check_ptr;
         }
```

Comment:
not related to the PR, but I find this a bit hard to read, maybe we could flatten like

```
let tx_type = EthTransaction.get_tx_type(tx_data);
let tx_type_0 = is_zero(tx_type);
let is_v_le_28 = is_le(v, 28);
let y_parity = tx_type_0 * ((v - 27) * is_v_le_28 + (v - 2 * chain_id - 35) * (1 - is_v_le_28)) + v * (1 - tx_type_0)
```

Reply:
or with
```
let offset = tx_type_0 * (is_v_le_28 * 27 + (1 - is_v_le_28) * (2 * chain_id + 35))
let y_parity = v = offset
```

Reply:
I think that is ok like that except `assert pre_eip155_tx = is_eip155_tx;` this line is confusing because in the middle when it should be at the end of the block as it is use only after in the code.
But open to flattening (I prefer second option)